### PR TITLE
images: fix environment variables duplication

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

remove the duplication of the environment variables

/assign @hasheddan @saschagrunert 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by cpanato** [comment]
my fault, overlook the variables 😞

File: images/build/go-runner/cloudbuild.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ steps:
     - REVISION=$_REVISION
     - GO_VERSION=$_GO_VERSION
     - DISTROLESS_IMAGE=$_DISTROLESS_IMAGE
-    - REGISTRY=$_REGISTRY
+
```
Comment (hasheddan) [inline images/build/go-runner/cloudbuild.yaml:28]: nitpick: did you mean to add newline here?

Comment (cpanato) [reply images/build/go-runner/cloudbuild.yaml:28]: to split the section, to make clear, I can revert if you want

Comment (cpanato) [reply images/build/go-runner/cloudbuild.yaml]: removed the blank line

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**